_adr/86/index.adoc

```diff
@@ -13,21 +13,87 @@ tags:
 ---
 ## Context and Problem Statement
 
+Canary tool is responsible to establish the health of the Kafka cluster. To learn more about the Strimzi Canary tool used in Kafka clusters, 
+please refer https://github.com/strimzi/strimzi-canary[here] . 
+
+Strimzi canary currently uses SASL/Plain (Simple Authentication and Security Layer - SASL) authentication mechanism in our Red Hat OpenShift Streams for Apache Kafka (RHOSAK) environment. Managed Application Services Single Sign On (MAS SSO) is used as the authentication service. As a part of the MAS SSO to sso.redhat.com migration, the existing Canary authentication mechanism will undergo some changes.
+
+The purpose of this ADR is to detail the existing authentication flow using MAS SSO and the authentication changes post migration to sso.redhat.com
+
 ## Goals
 
+* Secure Strimzi Canary authentication with sso.redhat.com 
+
 ## Non-goals
 
 ## Current architecture
 
+Currently, Kas Fleet Manager (KFM) creates a Canary service account client in MAS SSO upon a request to create and provision a Kafka cluster. These Canary service account clients are associated with the same organization as the user making the request to create a Kafka cluster. This information is contained in the `customClaimCheck` field in the Kafka Custom Resource (CR)
+
+```
+   - authentication:
+      customClaimCheck: '@.rh-org-id == ''testorgid''|| @.org_id == ''testorgid'''
+      tokenEndpointUri: >-
+            https://tokenendpointauri.com/token
```

Comment:
Just FYI [RFC-2606](https://www.rfc-editor.org/rfc/rfc2606.html) provides `example.com` for this kind of purpose.

Reply:
thanks Tom.. That is useful information.